java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.compare;
+
+import org.apache.arrow.vector.BaseFixedWidthVector;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Visitor to compare floating point.
+ */
+public class ApproxEqualsVisitor extends VectorEqualsVisitor {
+
+  private final float epsilon;
+
+  public ApproxEqualsVisitor(ValueVector right, float epsilon) {
```

Comment:
can you pls add a comment for the param ? eg. the float/double values are treated as equal as long as the delta is <= epsilon.

Reply:
Thanks, fixed.